test/units/TEST-07-PID1.file-write.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# shellcheck disable=SC2016
+set -eux
+set -o pipefail
+
+# shellcheck source=test/units/test-control.sh
+. "$(dirname "$0")"/test-control.sh
+# shellcheck source=test/units/util.sh
+. "$(dirname "$0")"/util.sh
+
+TEST_ID=TEST-07-PID1-file-write
+CONTAINER_ROOT_DIR="/tmp/${TEST_ID}-root"
+
+HOST_MOUNT_DIR=/tmp/test-dir
+INTERNAL_MOUNT_DIR="/${TEST_ID}"
+
+FILE_WR_SERVICE="${TEST_ID}.service"
+OUTPUT_FILE=test-service-output
+
+OUTPUT_CONTENTS='Test service is running'
+
+# Make test service that writes to a file
+internal_writer() {
+    # Create a test service that will run in the internal systemd
+    local container_systemd_dir="${CONTAINER_ROOT_DIR}/usr/lib/systemd/system"
+    local service_output="${INTERNAL_MOUNT_DIR}/${OUTPUT_FILE}"
+    local internal_test_service="${container_systemd_dir}/test-service.service"
+
+    mkdir -p "$container_systemd_dir"
+    cat <<EOF >"$internal_test_service"
+[Unit]
+Description=Test Service for Internal Systemd
+After=basic.target
+
+[Service]
+Type=oneshot
+ExecStart=/bin/sh -c 'echo "$OUTPUT_CONTENTS"  > "$service_output"'
+ExecStartPost=/usr/bin/systemctl --no-block exit 0
+TimeoutStopSec=15s
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+    systemctl --root="$CONTAINER_ROOT_DIR" enable test-service.service
+}
+
+testcase_multiple_features() {
+    local squashed_container_image=/usr/share/minimal_2.raw
+    local bind_mount_arg="${HOST_MOUNT_DIR}:${INTERNAL_MOUNT_DIR}"
+    local host_output="${HOST_MOUNT_DIR}/${OUTPUT_FILE}"
+
+    unsquashfs -no-xattrs -d "$CONTAINER_ROOT_DIR" "$squashed_container_image"
```

Comment:
Couldn't a tmpfs + bind mounting /usr/ or so be used instead? Building an image just on unpack it in /tmp seems a bit heavyweight